lib/services/image.ts

```diff
@@ -0,0 +1,63 @@
+import { s3 } from './s3';
+import sharp from 'sharp';
+import { v4 } from 'uuid';
+
+const AWS_S3_BUCKET_NAME = process.env.AWS_BUCKET_NAME_BOOKBAZAR as string;
+const IMAGE_WIDTH = 300;
+const IMAGE_HEIGHT = 400;
+
+export async function uploadImage(image: string): Promise<string>
+{
+  const buffer = await resizeImage(image);
+  const key: string = v4()
+  const params = {
+    Body: buffer, 
+    Bucket: AWS_S3_BUCKET_NAME, 
+    Key: key, 
+    ServerSideEncryption: 'AES256', 
+    StorageClass: 'STANDARD_IA',
+    ContentEncoding: 'base64',
+    ContentType: 'image/png'
+  };
+
+  return new Promise(function(resolve, reject) {
+    s3.putObject(params, function(err, data){
+      if (err) {
+        reject(err);
+      }
+      else {
+        const link = `https://${AWS_S3_BUCKET_NAME}.s3.us-east-2.amazonaws.com/${key}`;
+        resolve(link);
+      }
+    })
+  })
+}
+
+async function resizeImage(image: string): Promise<Buffer> {
```

Comment:
I think this should take a Buffer as input instead of a string. (Unless this is intentional?)